### ISSUE TITLE
Old Ubuntu 20.04 AMIs are used by default when provisioning downstream clusters in AWS

### ISSUE THREAD

**Issue by snasovich** [rancher/rancher#46241]
**Rancher Server Setup**
- Rancher version: v2.9-head / v2.8-head (a27b103859a84fe743e05d84c054659945b54e72)
- Installation option (Docker install/Helm Chart): Any
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: Any
- Cluster Type (Local/Downstream): Downstream, Amazon EC2 node driver provisioned
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom) Any
  - If custom, define the set of permissions:

**Describe the bug**
Old Ubuntu 20.04 AMIs are used by default when provisioning downstream clusters in AWS

**To Reproduce**
1. Setup AWS cloud credentials (if not setup)
2. Provision RKE1 or RKE2 cluster leaving all options by default (most importantly, leave "AMI ID" field blank)

**Result**
Nodes are provisioned with Ubuntu 20.04 while 22.04 is the only officially supported version per support matrix for recent releases (e.g. https://www.suse.com/suse-rancher/support-matrix/all-supported-versions/rancher-v2-8-5/).
Output from node:
```
ubuntu@(name):~$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.2 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.2 LTS"
VERSION_ID="20.04"
```

**Expected Result**
Ubuntu 22.04 is used as OS by default (unless specific AMI is specified by the user in "AMI ID" field)

**Screenshots**
N/A

**Additional context**
Default AMIs seems to be coming from https://github.com/rancher/machine/blob/6108a82e67bceaf7df9e0ff6dd541258369fba6a/drivers/amazonec2/region.go#L13
Updating it with 22.04 AMIs from https://cloud-images.ubuntu.com/locator/ec2/ should likely address this issue.

**Comment by jiaqiluo** [comment]
waiting for cutting a new rancher-machine RC and bringing it into rancher/rancher

**Comment by markusewalker** [comment]
**QA TEST PLAN**

|#|Scenario|
|---|---|
|1|Provision RKE1 downstream node driver w/AWS defaults|
|2|Provision RKE2 downstream node driver w/AWS defaults|
|3|Provision K3S downstream node driver w/AWS defaults|

**Comment by jiaqiluo** [comment]
PR is merged, now waiting for new r/r image

**Comment by jiaqiluo** [comment]
the fix can be validated on the latest v2.9-head

**Comment by markusewalker** [comment]
Validated that this issue, see details below:

**ENVIRONMENT DETAILS**
	- Rancher install: Docker
	- Rancher version: `v2.9-head` https://github.com/rancher/rancher/commit/97418c803ac1e5b7f28088559a20a97136f61021

**TEST RESULT**
|#|Scenario|Result
|---|---|---|
|1|Provision RKE1 downstream node driver w/AWS defaults|:white_check_mark:|
|2|Provision RKE2 downstream node driver w/AWS defaults|:white_check_mark:|
|3|Provision K3S downstream node driver w/AWS defaults|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. Provisioned downstream RKE1 `v1.30.3-rancher1-1` AWS node driver cluster.
     - Ensured that I used the default AMI.
2. Validated that cluster and nodes came up as `Active`.
3. Validated that the OS used is Ubuntu 22.04 for each node in the cluster.

_____

**Scenario 2**
1. Repeated scenario, but provisioned RKE2 `v1.30.2+rke2r1` AWS node driver cluster instead.

______

**Scenario 3**
Repeated scenario, but provisioned K3S `v1.30.2+k3s2` AWS node driver cluster instead.
